x-pack/plugins/ml/public/application/components/data_grid/use_column_chart.tsx

```diff
@@ -0,0 +1,442 @@
+    const id = stringHash(c.id);
+    aggs[id] = {
+      stats: {
+        field: c.id,
+      },
+    };
+    return aggs;
+  }, {} as Record<string, object>);
+
+  let respStats: any;
+  try {
+    respStats = await api.esSearch({
+      index: indexPatternTitle,
+      size: 0,
+      body: {
+        query,
+        aggs: minMaxAggs,
+        size: 0,
+      },
+    });
+  } catch (e) {
+    throw new Error(e);
+  }
+
+  return Object.keys(respStats.aggregations).reduce((p, aggName) => {
+    const stats = [respStats.aggregations[aggName].min, respStats.aggregations[aggName].max];
+    if (!stats.includes(null)) {
+      const delta = respStats.aggregations[aggName].max - respStats.aggregations[aggName].min;
+
+      let aggInterval = 1;
+
+      if (delta > MAX_CHART_COLUMNS) {
+        aggInterval = Math.round(delta / MAX_CHART_COLUMNS);
+      }
+
+      if (delta <= 1) {
+        aggInterval = delta / MAX_CHART_COLUMNS;
+      }
+
+      p[aggName] = { interval: aggInterval, min: stats[0], max: stats[1] };
+    }
+
+    return p;
+  }, {} as NumericColumnStatsMap);
+};
+
+interface AggHistogram {
+  histogram: {
+    field: string;
+    interval: number;
+  };
+}
+
+interface AggCardinality {
+  cardinality: {
+    field: string;
+  };
+}
+
+interface AggTerms {
+  terms: {
+    field: string;
+    size: number;
+  };
+}
+
+type ChartRequestAgg = AggHistogram | AggCardinality | AggTerms;
+
+export const fetchChartsData = async (
+  indexPatternTitle: string,
+  api: any,
+  query: any,
+  columnTypes: EuiDataGridColumn[]
+): Promise<ChartData[]> => {
+  const aggIntervals = await getAggIntervals(indexPatternTitle, api, query, columnTypes);
+
+  const chartDataAggs = columnTypes.reduce((aggs, c) => {
+    const fieldType = getFieldType(c.schema);
+    const id = stringHash(c.id);
+    if (fieldType === KBN_FIELD_TYPES.NUMBER || fieldType === KBN_FIELD_TYPES.DATE) {
+      if (aggIntervals[id] !== undefined) {
+        aggs[`${id}_histogram`] = {
+          histogram: {
+            field: c.id,
+            interval: aggIntervals[id].interval !== 0 ? aggIntervals[id].interval : 1,
+          },
+        };
+      }
+    } else if (fieldType === KBN_FIELD_TYPES.STRING || fieldType === KBN_FIELD_TYPES.BOOLEAN) {
+      if (fieldType === KBN_FIELD_TYPES.STRING) {
+        aggs[`${id}_cardinality`] = {
+          cardinality: {
+            field: c.id,
+          },
+        };
+      }
+      aggs[`${id}_terms`] = {
+        terms: {
+          field: c.id,
+          size: MAX_CHART_COLUMNS,
+        },
+      };
+    }
+    return aggs;
+  }, {} as Record<string, ChartRequestAgg>);
+
+  if (Object.keys(chartDataAggs).length === 0) {
+    return [];
+  }
+
+  let respChartsData: any;
+  try {
+    respChartsData = await api.esSearch({
+      index: indexPatternTitle,
+      size: 0,
+      body: {
+        query,
+        aggs: chartDataAggs,
+        size: 0,
+      },
+    });
+  } catch (e) {
+    throw new Error(e);
+  }
```

Comment:
same as above, throwing an error inside of the catch block

Reply:
Removed the whole try/catch block, errors will get caught by an outer try/catch anyway 82447ae.